radix-engine-interface/src/hooks.rs

```diff
@@ -17,7 +17,23 @@ pub struct OnDropInput {}
 pub type OnDropOutput = ();
 
 #[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
-pub struct OnMoveInput {}
+pub struct OnMoveInput {
+    /// True if the node moves from caller to callee, otherwise false.
+    pub is_moving_down: bool,
+
+    /// True if the destination actor is a barrier, otherwise false.
+    ///
+    /// TODO: expose generic information but fully-detailed actor?
+    pub is_to_barrier: bool,
+
+    /// True if the destination actor is auth zone, otherwise false.
+    ///
+    /// TODO: expose generic information but fully-detailed actor?
+    pub is_to_auth_zone: bool,
+
+    /// True if the destination actor is self blueprint.
+    pub is_to_self_blueprint: bool,
```

Comment:
A bit of a weird abstraction? Perhaps better would be exposing the `BlueprintId` of the destination actor?

Reply:
Yes, will do